File: src/main/java/nl/dannyjelsma/soundboardbot/music/Playlist.java
package nl.dannyjelsma.soundboardbot.music;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

@JsonIgnoreProperties(value = {"playlistFile"})
public class Playlist {

    private List<Song> songs = new ArrayList<>();
    private File playlistFile;

    public void addSong(Song song) {
        if (containsSong(song)) return;

        if (songs == null) {
            songs = new ArrayList<>();
        }

        songs.add(song);
    }

    public void removeSong(Song song) {
        songs.remove(song);
    }

    public boolean containsSong(Song song) {
        return songs.contains(song);
    }

    public List<Song> getSongs() {
        return songs;
    }

    public void setSongs(List<Song> songs) {
        this.songs = songs;
    }

    public File getPlaylistFile() {
        return playlistFile;
    }

    public void setPlaylistFile(File playlistFile) {
        this.playlistFile = playlistFile;
    }
}

File: src/main/java/nl/dannyjelsma/soundboardbot/music/Playlist.kt
package nl.dannyjelsma.soundboardbot.music

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.io.File

@JsonIgnoreProperties(value = ["playlistFile"])
class Playlist {
    var songs: MutableList<Song> = ArrayList()
    var playlistFile: File? = null

    fun addSong(song: Song) {
        if (containsSong(song)) return

        songs.add(song)
    }

    fun removeSong(song: Song) {
        songs.remove(song)
    }

    fun containsSong(song: Song): Boolean {
        return songs.contains(song)
    }
}